crates/matrix-sdk/src/widget/README.md

```diff
@@ -0,0 +1,74 @@
+# Widget Driver
+
+This module implements a widget driver. The widget driver allows a client to give widgets (webviews) access to Matrix via
+a specified `postMessage` API.
+It supports sending and reading events and provides some rudimentary client navigation features.
+There are some additional actions:
+
+- Get an OpenID token (not OAuth) to identify a user.
+- Ask for supported API versions.
+- Inform the client that the widget has loaded its content and is ready.
+
+## Usage
+
+To use this module, two structs are needed:
+
+- The `CapabilitiesProvider`: This trait uses the `acquire_capabilities` method to ask a client which capabilities
+  the widget is allowed. An example capability can be: reading events of type `"m.room.message"`.
+
+- The `WidgetDriver` itself, which consists of:
+  - The `driver`: Its only public method is `run`, which is used to actually start the widget communication.
+  - The message `handle`: This handle has `recv` and `send` methods.
+    A JSON message sent from the widget to the client needs to be passed to the handle via `send`.
+    Any JSON message the handle emits via `recv` needs to be sent to the widget via `postMessage`.
```

Comment:
As mentioned in our chat, it would be useful to make an ASCII graph that shows how these various components talk to each other and which messages get send over which channels.

(I know that this isn't talking about the internal divide we have with the machine and the driver, but I think it would still make sense to showcase this on the graph as well.)